controllers/cron_rollout_helpers.go

```diff
@@ -120,13 +126,29 @@ func createBaseDisruption(owner metav1.Object, disruptionSpec *chaosv1beta1.Disr
 }
 
 // setDisruptionAnnotations updates the annotations of a given Disruption object with those of its owner.
-// Additionally, it sets a scheduled time annotation using the provided scheduledTime.
-func setDisruptionAnnotations(disruption *chaosv1beta1.Disruption, owner metav1.Object, scheduledTime time.Time) {
-	for k, v := range owner.GetAnnotations() {
+// It sets a scheduled time annotation using the provided scheduledTime.
+// It parses the UserInfo annotation if it exists and sets user-related annotations.
+func setDisruptionAnnotations(disruption *chaosv1beta1.Disruption, owner metav1.Object, scheduledTime time.Time, log *zap.SugaredLogger) {
+	// Copy owner annotations to Disruption
+	ownerAnnotations := owner.GetAnnotations()
+	for k, v := range ownerAnnotations {
 		disruption.Annotations[k] = v
 	}
 
+	// Set scheduled time annotation
 	disruption.Annotations[ScheduledAtAnnotation] = scheduledTime.Format(time.RFC3339)
+
+	// Check if the UserInfo annotation exists and set user-related annotations
+	if userInfoJSON, exists := ownerAnnotations["UserInfo"]; exists {
+		var userInfo authv1beta1.UserInfo
+		if err := json.Unmarshal([]byte(userInfoJSON), &userInfo); err != nil {
+			log.Errorw("unable to parse UserInfo annotation", "err", err, "disruptionName", disruption.Name)
```

Comment:
```suggestion
			return errores.New("unable to unmarshal UserInfo: %w")
```
The code bellow will raise an error if the userInfo is not deserializable.